dev-lang/fpc/fpc-3.2.0.recipe

```diff
@@ -0,0 +1,159 @@
+	cmd:fpcjres$secondaryArchSuffix = $portVersion
+	cmd:fpclasschart$secondaryArchSuffix = $portVersion
+	cmd:fpclasschart.rsj$secondaryArchSuffix = $portVersion
+	cmd:fpcmake$secondaryArchSuffix = $portVersion
+	cmd:fpcmkcfg$secondaryArchSuffix = $portVersion
+	cmd:fpcmkcfg.rsj$secondaryArchSuffix = $portVersion
+	cmd:fpcres$secondaryArchSuffix = $portVersion
+	cmd:fpcsubst$secondaryArchSuffix = $portVersion
+	cmd:fpcsubst.rsj$secondaryArchSuffix = $portVersion
+	cmd:fpdoc$secondaryArchSuffix = $portVersion
+	cmd:fppkg$secondaryArchSuffix = $portVersion
+	cmd:fprcp$secondaryArchSuffix = $portVersion
+	cmd:h2pas$secondaryArchSuffix = $portVersion
+	cmd:h2paspp$secondaryArchSuffix = $portVersion
+	cmd:instantfpc$secondaryArchSuffix = $portVersion
+	cmd:makeskel$secondaryArchSuffix = $portVersion
+	cmd:mkarmins$secondaryArchSuffix = $portVersion
+	cmd:mkinsadd$secondaryArchSuffix = $portVersion
+	cmd:mkx86ins$secondaryArchSuffix = $portVersion
+	cmd:pas2fpm$secondaryArchSuffix = $portVersion
+	cmd:pas2jni$secondaryArchSuffix = $portVersion
+	cmd:pas2ut$secondaryArchSuffix = $portVersion
+	cmd:pas2ut.rsj$secondaryArchSuffix = $portVersion
+	cmd:plex$secondaryArchSuffix = $portVersion
+	cmd:postw32$secondaryArchSuffix = $portVersion
+	cmd:ppdep$secondaryArchSuffix = $portVersion
+	cmd:ppudump$secondaryArchSuffix = $portVersion
+	cmd:ppufiles$secondaryArchSuffix = $portVersion
+	cmd:ppumove$secondaryArchSuffix = $portVersion
+	cmd:ptop$secondaryArchSuffix = $portVersion
+	cmd:pyacc$secondaryArchSuffix = $portVersion
+	cmd:rmcvsdir$secondaryArchSuffix = $portVersion
+	cmd:rstconv$secondaryArchSuffix = $portVersion
+	cmd:rstconv.rsj$secondaryArchSuffix = $portVersion
+	cmd:unitdiff$secondaryArchSuffix = $portVersion
+	cmd:unitdiff.rsj$secondaryArchSuffix = $portVersion
+	cmd:cvsco.tdf$secondaryArchSuffix = $portVersion
+	cmd:cvsdiff.tdf$secondaryArchSuffix = $portVersion
+	cmd:cvsup.tdf$secondaryArchSuffix = $portVersion
+	cmd:fp.ans$secondaryArchSuffix = $portVersion
+	cmd:gplprog.pt$secondaryArchSuffix = $portVersion
+	cmd:gplunit.pt$secondaryArchSuffix = $portVersion
+	cmd:grep.tdf$secondaryArchSuffix = $portVersion
+	cmd:makeskel.rsj$secondaryArchSuffix = $portVersion
+	cmd:program.pt$secondaryArchSuffix = $portVersion
+	cmd:ptop.rsj$secondaryArchSuffix = $portVersion
+	cmd:tpgrep.tdf$secondaryArchSuffix = $portVersion
+	cmd:unit.pt$secondaryArchSuffix = $portVersion
+	cmd:json2pas$secondaryArchSuffix = $portVersion
+	cmd:mka64ins$secondaryArchSuffix = $portVersion
+	cmd:pas2js$secondaryArchSuffix = $portVersion
+	cmd:webidl2pas$secondaryArchSuffix = $portVersion
+	lib:libpas2jslib$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix  >= 6.0
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel >= r1~alpha4_pm_hrev51411
+	"
+BUILD_PREREQUIRES="
+	binutils$secondaryArchSuffix
+	cmd:tar
+	cmd:make
+	cmd:diff
+	cmd:find
+	cmd:which
+	cmd:gcc
```

Comment:
If this is meant to be used on a secondary architecture (gcc8), shouldn't this be "cmd:gcc$secondaryArchSuffix"?

Reply:
Freepascal itself could be compiled either in a gcc2 or in a gcc8 environment (by default, it is gcc2). 

As the compiler compile itself, gcc is not used at all. This is the reason for the use of a bootstrap package, that only convert the default freepascal distribution into an Haiku package containing the corresponding binaries. A specific freepascal version is only guaranteed to be compiled using the previous official one. 

Only the link is done using ld, in the defined environment. The only adjustment that might be required is the default linking paths between the two environments. More informations on this point is here : https://bugs.freepascal.org/view.php?id=26486. But not easily solved in fpc itself. Maybe with a patch in haikuport ? 

Compiling in a gcc8 environment would enable building Lazarus which link with Qt libraries, obviously not available in the gcc2 environment.

Reply:
> Compiling in a gcc8 environment

This would mean that gcc$secondaryArchSuffix is needed then?
PS, checking a build for that works ok for this recipe (with a few changes), will report later

Reply:
So far managed to create a _x86 package (with gcc8), will push my work to my github account and link to it in a bit here.
The outcome for "fpc-x86" in Terminal: https://paste.ubuntu.com/p/Qxh7JBmfTc/ (link will stay up for a month)